Source/Core/Core/ScriptEngine.cpp

```diff
@@ -0,0 +1,945 @@
+// Copyright 2008 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+// ScriptEngine
+// Expands the emulator using LuaJIT scripts.
+
+#include "Core/ScriptEngine.h"
+
+#include <limits>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+#include <cassert>
+#include <lua.hpp>
+
+#include "Common/Assert.h"
+#include "Common/IniFile.h"
+
+#include "Core/ConfigManager.h"
+#include "Core/PowerPC/MMU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+// clang-format off
+#define DOLPHIN_LUA_METHOD(x) {.name = #x, .func = (x)}
+// clang-format on
+
+namespace ScriptEngine
+{
+// Namespace Lua hosts static binding code.
+namespace Lua
+{
+static const char* s_context_registry_key = "c";
```

Comment:
```suggestion
static constexpr char s_context_registry_key[] = "c";
```

Will only save the string data into the binary rather than a pointer as well.